### DISCUSSION TITLE
Force displaying am/pm for times in the en-GB locale?

### DISCUSSION THREAD

**Discussion by mjb8086** [moment/luxon#1629] [Q&A] [answered]
Hi, using Luxon to pretty print ISO 8601 dates in my application.

I have tried both `.toLocaleString(DateTime.DATETIME_MED)` and `.toFormat("DD t")`. While both options give the date correctly, e.g. as `15 May 2024` the time is given in 24 hours. In my locale (GB) it is more common to refer to the time as am/pm. I can see from the dt.locale property the locale is en-GB. Yet even when I use the explicit format sting for 12 hour time, 24 hour time is all I get.

How can I force printing of 12 hour time? I'd like this to be at least an option in my application.

Best, MJB

**Comment by icambron** [comment, marked as answer]
Sure:

```js
DateTime
     .now()
     .setLocale("en-GB")
     .toLocaleString({...DateTime.DATETIME_MED, "hourCycle": "h11"}) // => '15 May 2024, 1:40 pm'
```

The argument to `toLocaleString()` is ultimately passed to the `Intl.DateTimeFormat` constructor and thus supports everything that does. `DateTime.DATETIME_MED` is just a precanned object containing formatting options. So you can just glom on extra options, as in my example.

Note that you have to override it not because of anything about `DATETIME_MED` but actually because of the `en-GB` locale, which does specify 24-hour time. In my locale ,`en-US`, there's no debate and we always use 12-hour time, so the locale works like that by default. The UK seems to be a bit of an in-betweener on this with official stuff in 24-hour time and informal stuff in am/pm, which puts the ICU folks in a bind about what to make the locale do.

**Comment by mjb8086** [comment]
@icambron Thanks for the swift reply, that got me sorted. Given that we also purchase fuel by the litre while measuring fuel efficiency in miles per gallon, I can understand the ICU's predicament.
